Content/Items/Food/Special/Bonus.ShrimpSandwich.cs

```diff
@@ -17,12 +24,295 @@ public override FoodRecipie Recipie()
+	{
+		public bool Active = false;
+
+		//reusing this modplayer since its related and so im not making one just for 1 method
+		public override void AnglerQuestReward(float rarityReduction, List<Item> rewardItems)
+		{
+			if (rarityReduction > 0.45f && Main.rand.NextBool(4, 5))//skip if the reward is too rare, or at random sometimes
+			{
+				var shrimp = new Item();
+				shrimp.SetDefaults(ModContent.ItemType<JumboShrimp>());
+				shrimp.stack = Main.rand.Next(2, 6);//copied fish potion chances
+				rewardItems.Add(shrimp);
+			}
+		}
+
+		public override void ResetEffects()
+		{
+			Active = false;
+		}
+	}
+
+	public class ShrimpSandwhichDropsNPC : GlobalNPC
+	{
+		private static MethodInfo RollFishingDropInfo;
+		public delegate void RollFishingDropDelegate(Projectile proj, ref FishingAttempt fisher);
+		public static RollFishingDropDelegate RollFishingDrop;
+
+		public override void Load()
+		{
+			RollFishingDropInfo = typeof(Terraria.Projectile).GetMethod("FishingCheck_RollItemDrop", BindingFlags.NonPublic | BindingFlags.Instance);
+			RollFishingDrop = (RollFishingDropDelegate)Delegate.CreateDelegate(typeof(RollFishingDropDelegate), RollFishingDropInfo);
+		}
+
+		public override void Unload()
+		{
+			RollFishingDrop = null;
+		}
+
+		public override void ModifyGlobalLoot(GlobalLoot globalLoot)
+		{
+			var leadingConditionRule = new LeadingConditionRule(new ShrimpSandwhichDropsCondition());
+			leadingConditionRule.OnSuccess(new FishDrop(8));
+			globalLoot.Add(leadingConditionRule);
+			base.ModifyGlobalLoot(globalLoot);
+		}
+	}
+
+	public class FishDrop : IItemDropRule
+	{
+		public int chanceDenominator;
+		public int chanceNumerator;
+
+		public List<IItemDropRuleChainAttempt> ChainedRules { get; private set; }
+
+		public FishDrop(int chanceDenominator, int chanceNumerator = 1)
+		{
+			this.chanceDenominator = chanceDenominator;
+			this.chanceNumerator = chanceNumerator;
+			ChainedRules = [];
+		}
+
+		public virtual bool CanDrop(DropAttemptInfo info)
+		{
+			return true;
+		}
+
+		//check copied from FishingCheck() in projectile.cs
+		private int GetHeightTeir(int TilePositionY)
+		{
+			if (TilePositionY < Main.worldSurface * 0.5)
+			{
+				return 0;//space
+			}
+			else if (TilePositionY < Main.worldSurface)
+			{
+				return 1;//overworld
+			}
+			else if (TilePositionY < Main.rockLayer)
+			{
+				return 2;//dirt layer
+			}
+			else if (TilePositionY < Main.maxTilesY - 300)
+			{
+				return 3;//rock layer
+			}
+			else
+			{
+				return 4;//underworld
+			}
+		}
+
+		//modifed from FishingCheck_RollDropLevels() in projectile.cs, removed crate logic
+		private void RollDropLevels(int fishingLevel, out bool common, out bool uncommon, out bool rare, out bool veryrare, out bool legendary)
+		{
+			int num = 150 / fishingLevel;
+			int num2 = 300 / fishingLevel;
+			int num3 = 1050 / fishingLevel;
+			int num4 = 2250 / fishingLevel;
+			int num5 = 4500 / fishingLevel;
+
+			if (num < 2)
```

Comment:
Is there a chance we can rename these variables from adapted vanilla code to make it a bit more palletable to read?